worker/logger/logger.go

```diff
@@ -43,6 +46,12 @@ func NewLogger(api *logger.State, agentConfig agent.Config) (worker.Worker, erro
 
 func (logger *Logger) setLogging() {
 	loggingConfig, err := logger.api.LoggingConfig(logger.agentConfig.Tag())
+	if logger.configOverride != "" {
+		log.Debugf("overriding model config %q with override from agent.conf %q", loggingConfig, logger.configOverride)
+		loggingConfig = logger.configOverride
+		// If we are overriding the logging config value, there is no error.
```

Comment:
Why not do this block fist. Set loggingConfig to logger.configOverride and then if empty, read from api. Will be much simpler IMO.